src/ui/public/elasticsearch_errors/is_term_size_zero_error.js

```diff
@@ -1,6 +1,10 @@
 import ElasticsearchError from './elasticsearch_error';
 
 export default function isTermSizeZeroError(error) {
-  const esError = new ElasticsearchError(error);
-  return esError.hasRootCause('size must be positive, got 0');
+  try {
+    const esError = new ElasticsearchError(error);
+    return esError.hasRootCause('size must be positive, got 0');
+  } catch (err) {
+    return false;
+  }
```

Comment:
I think the fact that `ElasticsearchError` asserts the error type during instantiation is absolutely the correct behavior.  It should be impossible to create an ElasticsearchError with something that isn't a known acceptable state of a real Elasticsearch error.  You should not be able to instantiate *any* class in an unacceptable/unexpected state.

The crux of the issue for me is that exposing this functionality in a stateful class makes ES error handling unnecessarily restrictive.  Consider if instead of a class, we exposed a module with the following:

```
getRootCauses(error)
hasRootCause(error, cause)
isValid(error)
```

Then you can compose those functions to produce other useful functions in an expressive way:

```
isTermSizeZeroError(error) {
  return isValid(error) && hasRootCause('size must be positive, got 0');
}
```

At no time did that involve sacrificing the integrity of any state (it's stateless) or wrapping an implementation detail of a constructor in a procedure in order to produce some slightly different behavior as a new function.

Plus, if there's still good reason to have the ElasticsearchError class elsewhere, then you can always compose the public class API from those same stateless functions.